File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/FreeCameraOptions.java
package org.maplibre.nativeffi;

import java.util.Objects;

/** Mutable descriptor for free-camera position and orientation. */
public final class FreeCameraOptions {
  private Vec3 position;
  private Quaternion orientation;

  public boolean hasPosition() {
    return position != null;
  }

  public Vec3 position() {
    return position;
  }

  public FreeCameraOptions setPosition(Vec3 position) {
    this.position = Objects.requireNonNull(position, "position");
    return this;
  }

  public FreeCameraOptions clearPosition() {
    position = null;
    return this;
  }

  public boolean hasOrientation() {
    return orientation != null;
  }

  public Quaternion orientation() {
    return orientation;
  }

  public FreeCameraOptions setOrientation(Quaternion orientation) {
    this.orientation = Objects.requireNonNull(orientation, "orientation");
    return this;
  }

  public FreeCameraOptions clearOrientation() {
    orientation = null;
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/camera/FreeCameraOptions.kt
package org.maplibre.nativeffi.camera

import org.maplibre.nativeffi.geo.Quaternion
import org.maplibre.nativeffi.geo.Vec3

/** Mutable free-camera descriptor. */
public class FreeCameraOptions {
  public var position: Vec3? = null

  public var orientation: Quaternion? = null
}
